proposals/csharp-7.0/tuples.md

```diff
@@ -0,0 +1,437 @@
+
+A well formed "big tuple" will have names `Item1` etc. all the way up to the number of tuple elements, even though the underlying type doesn't physically have those fields directly defined. The same goes for the tuple returned from the `Rest` field, only with the numbers "shifted" appropriately.
+
+For tuple element names occurring in partial type declarations, the names must be the same.
+
+```csharp
+partial class C : IEnumerable<(string name, int age)> { ... }
+partial class C : IEnumerable<(string fullname, int)> { ... } // error: names must be specified and the same
+```
+
+When tuple elements names are used in overridden signatures, or implementations of interface methods, tuple element names in parameter and return types must be preserved. It is an error for the same generic interface to be inherited or implemented twice with identity convertible type arguments that have conflicting tuple element names
+
+> note:
+>
+> For the purpose of overloading, overriding and hiding, tuples of the same types and lengths as well as their underlying ValueTuple types are considered equivalent. All other differences are immaterial. When overriding a member it is permitted to use tuple types with same or different field names than in the base member.
+>
+> A situation where same field names are used for non-matching fields between base and derived member signatures, a warning is reported by the compiler.  
+
+```csharp
+class Base
+{
+    virtual void M1(ValueTuple<int, int> arg){...}
+}
+class Derived : Base
+{
+    override void M1((int c, int d) arg){...} // valid override, signatures are equivalent
+}
+class Derived2 : Derived 
+{
+    override void M1((int c1, int c) arg){...} // also valid, warning on possible misuse of name 'c' 
+}
+
+class InvalidOverloading 
+{
+    virtual void M1((int c, int d) arg){...}
+    virtual void M1((int x, int y) arg){...}        // invalid overload, signatures are eqivalent
+    virtual void M1(ValueTuple<int, int> arg){...}  // also invalid
+}
+```
+
+> endnote.
+
+### Tuple field name erasure at runtime
+
+Tuple field names aren't part of the runtime representation of tuples, but are tracked only by the compiler. As a result, the field names will not be available to a 3rd party observer of a tuple instance - such as reflection or dynamic code.
+
+In alignment with the identity conversions, a boxed tuple does not retain the names of the fields and will unbox to any tuple type that has the same element types in the same order.
+
+```csharp
+object o = (a: 1, b: 2);           // boxing conversion
+var t = ((int moo, int boo))o;     // unboxing conversion
+```
+
+## Additions to [Variables](../../spec/variables.md)
+
+> The following text should be added at the end of the [Variables](../../spec/variables.md) section.
+
+### "Discards"
+
+The `_` can be used as a *discard* under these rules:
```

Comment:
>used as a *discard* [](start = 15, length = 19)

I don't know what "used as a *discard*" means.

Reply:
logged in #3181